osu.Game/Overlays/BeatmapSetOverlay.cs

```diff
@@ -85,10 +87,16 @@ public BeatmapSetOverlay()
         }
 
         [BackgroundDependencyLoader]
-        private void load(APIAccess api, RulesetStore rulesets)
+        private void load(APIAccess api, RulesetStore rulesets, BeatmapManager manager)
         {
             this.api = api;
             this.rulesets = rulesets;
+
+            manager.BeatmapSetAdded += beatmap =>

```

Comment:
this is never unbound. also i'm not sure i like this logic – why should it hide?

Reply:
was an arbitrary decision based on the fact that the DirectPanel the overlay came from is removed. seemed like either that or remove the download buttons on completion